### PR TITLE
rename touchpad to gesture, and add new gestures

### PR DESCRIPTION
**Pull request by mockersf**
# Objective

- With the recent winit update, touchpad specific events can also be triggered on mobile

## Solution

- Rename them to gestures and add support for the new ones

## Testing

- Tested on the mobile example on iOS

https://github.com/bevyengine/bevy/assets/8672791/da4ed23f-ff0a-41b2-9dcd-726e8546bef2

## Migration Guide

- `TouchpadMagnify` has been renamed to `PinchGesture`
- `TouchpadRotate` has been renamed to `RotationGesture `

### CODE REVIEW

File: crates/bevy_input/src/gestures.rs [hunk 1]
```diff
@@ -0,0 +1,73 @@
+//! Gestures functionnality, from touchscreens and touchpads.
```
Comment (alice-i-cecile) [inline crates/bevy_input/src/gestures.rs:1]:
```suggestion
//! Gestures functionality, from touchscreens and touchpads.
```

**Review by alice-i-cecile — APPROVED** [review]
Good improvements. I am not thrilled about the platform-specific limitations, but this is an incremental improvement nonetheless.

**Comment by mockersf** [comment]
there's a PR to add some of them to Wayland https://github.com/rust-windowing/winit/pull/3656, and I started looking for Android

File: crates/bevy_window/src/window.rs [hunk 2]
```diff
@@ -281,6 +281,33 @@ pub struct Window {
...
+    /// - On macOS, they are recognized by default and can't be disabled.
+    pub recognize_rotation_gesture: bool,
+    /// Sets whether this window recognize [`DoubleTapGesture`]
+    ///
+    /// ## Platform-specific
+    ///
+    /// - Only used on iOS.
+    /// - On macOS, they are recognized by default and can't be disabled.
+    pub recognize_doubletap_gesture: bool,
+    /// Sets whether this window recognize [`PanGesture`], with a number of fingers between the first value and the last.
```
[the remark this answers is not in the document]
Comment (mockersf) [reply crates/bevy_window/src/window.rs]: it's the number of fingers you use at the same time for the gesture, so (0, 5) would detect pan gesture with 0, 1, 2, 3, 4 or 5 fingers.

**Pull request merged**